include/jemalloc/internal/mutex_types.h

```diff
@@ -1,31 +1,64 @@
 #ifndef JEMALLOC_INTERNAL_MUTEX_TYPES_H
 #define JEMALLOC_INTERNAL_MUTEX_TYPES_H
 
+typedef struct lock_prof_data_s lock_prof_data_t;
 typedef struct malloc_mutex_s malloc_mutex_t;
+typedef struct malloc_mutex_noprof_s malloc_mutex_noprof_t;
+
+#ifdef _WIN32
+#  if _WIN32_WINNT >= 0x0600
+#    define MALLOC_MUTEX_LOCK    AcquireSRWLockExclusive(&mutex->lock)
```

Comment:
Overall this is a nice abstraction, though I would have a slightly easier time understanding uses of the macros if they were e.g.
```
#    define MALLOC_MUTEX_LOCK(m)    AcquireSRWLockExclusive(&(m)->lock)
```
with calls like:
```
MALLOC_MUTEX_LOCK(mutex);
```